apollo-federation/src/sources/connect/json_selection/parameter_extraction.rs

```diff
@@ -0,0 +1,102 @@
+use std::collections::HashSet;
+
+use super::JSONSelection;
+use super::NamedSelection;
+use super::PathSelection;
+use super::SubSelection;
+
+/// A representation of a static parameter.
+///
+/// Each parameter can include path components for drilling down into specific
+/// members of a parameter.
+///
+/// e.g. A parameter like below
+/// ```json_selection
+/// $this.a.b.c
+/// ```
+///
+/// would have a name of "this" and a path of ["a", "b", "c"].
+#[derive(Debug, Hash, PartialEq, Eq)]
+pub struct StaticParameter<'a> {
+    /// The name of the parameter, after the $
+    /// TODO: This might be nice to have as an enum, but it requires making
+    /// extraction fallible. Another option would be to have JSONSelection aware
+    /// of which variables it knows about, but that might not make sense to have
+    /// as a responsibility of JSONSelection.
+    pub name: &'a str,
+
+    /// Any paths after the name
+    pub paths: Vec<&'a str>,
+}
+
+pub trait ExtractParameters {
+    /// Extract parameters for static analysis
+    fn extract_parameters(&self) -> Option<HashSet<StaticParameter>>;
+}
+
+impl ExtractParameters for JSONSelection {
+    fn extract_parameters(&self) -> Option<HashSet<StaticParameter>> {
+        match &self {
+            JSONSelection::Named(named) => named.extract_parameters(),
+            JSONSelection::Path(path) => path.extract_parameters(),
+        }
+    }
+}
+
+impl ExtractParameters for PathSelection {
+    fn extract_parameters(&self) -> Option<HashSet<StaticParameter>> {
+        let param = match &self {
+            PathSelection::Var(name, rest) => Some(StaticParameter {
+                name: name.as_str(),
+                paths: rest
+                    .collect_paths()
+                    .iter()
+                    // We don't run `to_string` here since path implements display and prepends
+                    // a '.' to the path components
+                    .map(|k| match k {
+                        super::Key::Field(val) | super::Key::Quoted(val) => val.as_str(),
+                        super::Key::Index(_) => todo!(),
```

Comment:
what should happen here?

Reply:
@lennyburdette This `Key::Index` variant is going away in #5762, specifically in commit https://github.com/apollographql/router/pull/5762/commits/01bb990b1903a62d8165bb0c0d9af3559987b4a0.

Reply:
I've changed it to return a `[]` for now until the PR mentioned above gets merged.